### PR TITLE
customrules:chore - improve tests asserts and code cleaning

### PR DESCRIPTION
**Pull request by matheusalcantarazup**
Updates #718 

Signed-off-by: Matheus Alcantara <matheus.alcantara@zup.com.br>

**- What I did**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: internal/services/custom_rules/service.go [hunk 1]
```diff
@@ -60,55 +71,48 @@ func (s *Service) setCustomRules() {
...
 
-	s.customRulesByTool[customRules[index].Language] = append(
-		s.customRulesByTool[customRules[index].Language], s.parseCustomRuleToTextRule(index, customRules),
+	s.customRules[rule.Language] = append(
+		s.customRules[rule.Language], s.parseCustomRuleToTextRule(rule),
 	)
 }
 
-func (s *Service) openCustomRulesJSONFile() (customRules []customRulesEntities.CustomRule, err error) {
+func (s *Service) openCustomRulesJSONFile() (customRules []customrules.CustomRule, err error) {
```
Comment (nathanmartinszup) [inline internal/services/custom_rules/service.go:92]: func (s *Service) openCustomRulesJSONFile() (customRules []*customrules.CustomRule, err error) {
...

File: internal/services/custom_rules/service.go [hunk 2]
```diff
@@ -60,55 +71,48 @@ func (s *Service) setCustomRules() {
 	}
 
 	for index := range customRules {
-		s.validateAndParseCustomRule(index, customRules)
+		s.validateAndParseCustomRule(&customRules[index])
```
Comment (nathanmartinszup) [inline internal/services/custom_rules/service.go:74]: for index := range customRules {
		s.validateAndParseCustomRule(customRules[index])
	}

File: internal/services/custom_rules/service.go (hunk 1 above)
Comment (matheusalcantarazup) [reply internal/services/custom_rules/service.go]: Done.

File: internal/services/custom_rules/service.go (hunk 2 above)
Comment (matheusalcantarazup) [reply internal/services/custom_rules/service.go]: Done.

**Review by iancardosozup — APPROVED** [review]

**Review by nathanmartinszup — APPROVED** [review]

**Pull request merged**